### ISSUE TITLE
Bug: bad_msg error fails lambda

### ISSUE THREAD

**Issue by erikerikson** [Nordstrom/hello-retail#61]
In the copy-pasted code processing kinesis events, there is a bug in the complete callback.

The code is currently:
```
        let successes = 0
        const complete = (err) => {
          if (err) {
            console.log(err)
            // TODO uncomment following
            // throw new Error(`${constants.MODULE} ${err}`);
            // TODO remove rest of block to use above.
            const msg = `${constants.MODULE} ${err}`
            if (msg.indexOf(`${constants.MODULE} ${constants.METHOD_PROCESS_EVENT} ${constants.BAD_MSG}`) !== -1) {
              console.log('######################################################################################')
              console.log(msg)
              console.log('######################################################################################')
              successes += 1
            } else {
              throw new Error(msg)
            }
          } else {
            successes += 1
            if (successes === kinesisEvent.Records.length) {
              console.log(`${constants.MODULE} ${constants.METHOD_PROCESS_KINESIS_EVENT} - all ${kinesisEvent.Records.length} events processed successfully.`)
              callback(null, true)
            }
          }
        }
```

If a bad message is the last message of a bundle then the lambda will not complete properly, in an infinite loop.  The code should instead be as follows (note the movement of the `if (successes === kinesisEvent.Records.length)` conditional, a minor change is `callback(null, true)` to `callback()`):
```
        let successes = 0
        const complete = (err) => {
          if (err) {
            console.log(err)
            // TODO uncomment following
            // throw new Error(`${constants.MODULE} ${err}`);
            // TODO remove rest of block to use above.
            const msg = `${constants.MODULE} ${err}`
            if (msg.indexOf(`${constants.MODULE} ${constants.METHOD_PROCESS_EVENT} ${constants.BAD_MSG}`) !== -1) {
              console.log('######################################################################################')
              console.log(msg)
              console.log('######################################################################################')
              successes += 1
            } else {
              throw new Error(msg)
            }
          } else {
            successes += 1
          }
          if (successes === kinesisEvent.Records.length) {
            console.log(`${constants.MODULE} ${constants.METHOD_PROCESS_KINESIS_EVENT} - all ${kinesisEvent.Records.length} events processed successfully.`)
            callback()
          }
        }
```

**Comment by erikerikson** [comment]
this is my fault and it is systemically present.  that said, when we produce the module that simplifies/automates this handling, all the buggy code will be replaced so it's probably not a  priority.

**Comment by erikerikson** [comment]
Fixed via #66
